OpenRA.Mods.Common/Widgets/Logic/Ingame/ReplayControlBarLogic.cs

```diff
@@ -46,17 +46,17 @@ public ReplayControlBarLogic(Widget widget, World world, OrderManager orderManag
 				var originalTimestep = world.Timestep;
 
 				var pauseButton = widget.Get<ButtonWidget>("BUTTON_PAUSE");
-				pauseButton.IsVisible = () => world.ReplayTimestep != 0 && orderManager.NetFrameNumber < replayNetTicks;
+				pauseButton.IsVisible = () => world.ReplayTimestep != 0 && !world.PauseStateLocked && orderManager.NetFrameNumber < replayNetTicks;
 				pauseButton.OnClick = () => world.ReplayTimestep = 0;
 
 				var playButton = widget.Get<ButtonWidget>("BUTTON_PLAY");
-				playButton.IsVisible = () => world.ReplayTimestep == 0 || orderManager.NetFrameNumber >= replayNetTicks;
+				playButton.IsVisible = () => world.ReplayTimestep == 0 || world.PauseStateLocked || orderManager.NetFrameNumber >= replayNetTicks;
 				playButton.OnClick = () => world.ReplayTimestep = (int)Math.Ceiling(originalTimestep * multipliers[speed]);
-				playButton.IsDisabled = () => orderManager.NetFrameNumber >= replayNetTicks;
+				playButton.IsDisabled = () => world.PauseStateLocked || orderManager.NetFrameNumber >= replayNetTicks;
```

Comment:
These changes don't look correct - The `replayNetTicks` check is supposed to handle this, so that should either be fixed or removed.

Reply:
It isn't supposed to, `replayNetTicks` is a cached variable set to last replay tick. Going out of sync has nothing to do with replay parsing or replay length